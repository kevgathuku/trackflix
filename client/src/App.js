import React, { Component } from 'react';
import './App.css';

import MovieList from './components/MovieList';

const movies = [{

}];

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Trending TV Shows</h1>
        </header>
        <section>
          <MovieList />
        </section>
      </div>
    );
  }
}

export default App;
