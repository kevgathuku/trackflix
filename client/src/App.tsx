import * as React from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";

import { ShowProps } from "./types";
import { AppState } from "./reducer";
import MovieList from "./components/MovieList";
import { popularTvShowsFetch } from "./actions";
import "./App.css";

type AppProps = {
  fetchTvShows: Function;
  shows: Array<ShowProps>;
};

export class App extends React.Component<AppProps> {
  componentWillMount() {
    this.props.fetchTvShows();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Trending TV Shows</h1>
        </header>
        <section>
          <MovieList shows={this.props.shows} />
        </section>
      </div>
    );
  }
}

const mapStateToProps = (state: AppState) => {
  return {
    shows: state.discoveredShows
  };
};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchTvShows: () => dispatch(popularTvShowsFetch())
});

// If the mapDispatchToProps function is not provided,
// the default implementation just injects dispatch into your componentâ€™s props.
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
