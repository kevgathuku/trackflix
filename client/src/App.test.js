import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import { App } from "./App";

it("renders without crashing", () => {
  const tree = shallow(<App />);
  expect(toJSON(tree)).toMatchSnapshot();
});
