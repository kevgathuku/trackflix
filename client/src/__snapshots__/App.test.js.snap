// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Trending TV Shows
    </h1>
  </header>
  <section>
    <MovieList />
  </section>
</div>
`;
